Update Quality Declarations and READMEs

JLBuenoLopez-eProsima opened the pull request:
* Update name to Fast DDS instead of Fast-RTPS
* Provide links to the Quality Declarations for the external dependencies
* Add Quality Declaration for `rmw_fastrtps_dynamic_cpp`
* Add README for `rmw_fastrtps_dynamic_cpp`

Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

JLBuenoLopez-eProsima commented:
This pull request should cancel currently opened pull requests #360 and #357. The Quality Declarations provided are more up-to-date

JLBuenoLopez-eProsima commented:
Hi @Blast545 @ahcorde @brawner ,

Please review this pull request that substitutes current pull requests #360 and #357. I am not allowed to include reviewers into the pull request, so I leave you this comment.

brawner commented on README.md:
```diff
@@ -78,3 +78,11 @@ The following example configures Fast-RTPS to publish synchronously, and to have
+
+## Quality Declaration files
+
+Quality Declarations for the external dependencies of these packages can be found in:
```
We've been stating the quality declaration in the README as well. See https://github.com/ros2/rmw/blob/master/rmw/README.md for an example

JLBuenoLopez-eProsima commented:
I have updated the README for each package to fix the new Quality Level. I have also updated the library name to Fast DDS in those README.

I have provided the Quality Declaration and README for `rmw_fastrtps_dynamic_cpp This package is not included in the coverage job so its current Quality Level should be Q3. Also, the Quality Level of its dependencies `rosidl_typesupport_introspection_c` and `rosidl_typesupport_introspection_cpp` should be assessed.

Finally, I think some update work has to be done in several `rosidl_typesupport` packages. In some of them, the Quality Level stated in the README does not match the one in the Quality Declaration. Additionally, I think some of those packages could claim to be Quality Level 2 now that Fast DDS claims to be in that level.

Blast545 reviewed (commented):
LGTM, a couple of minor comments, I'll let @brawner @ahcorde give an extra review before approval

JLBuenoLopez-eProsima commented on rmw_fastrtps_dynamic_cpp/QUALITY_DECLARATION.md:
```diff
@@ -0,0 +1,203 @@
+* `rosidl_typesupport_introspection_c`
+* `rosidl_typesupport_introspection_cpp`
+
+The Quality Declarations for the last two dependencies are not available currently.
```
I have also added the links in the other two Quality Declarations

hidmic commented:
@JaimeMartin any reason why this PR got plain merged instead of squash-merged? We usually do the latter to keep repository history and CHANGELOGs clean, unless there's a powerful reason not to.

JaimeMartin commented:
@hidmic Sorry, it was a mistake.
